# Seattle Kids in Urban Centers

## Problem Statement:

## Background
Welcome to Seattle, the big city with a small town feeling.  Though Seattle is by all acounts a major metropolitan area, the small neighborhood focus of the city lends to a small town feeling.  These higher-density urban areas within the city are not an accident but rather an approach to focusing development in specific areas of the city, along corridors with transportation, to avoid rezoning large fractions of residential land. <cite> Today, most of the residential land in the city remains zoned exclusively for single-family homes.  <cite>

In July 2019, the city enacted legislation allowing the construction of accessory dwelling units (ADUs) on single-family zoned property. <cite> This change was enacted to help with a housing affordability crisis the city was facing.  The housing crisis continues despite the increase in ADUs and the city is currently examining zoning restrictions in all areas of the city on a neighborhood-by-neighborhood basis.  This rezoning has focused on the urban centers and villages 

The city of Seattle has seen population growth over the last twenty years.  While most residential land in the city has been zoned for single-family housing the explosive growth of the last two decades has resulted in significant growth in more urban areas of the city.  The city designates these areas as 32 individual 'Urban Centers and Villages.'  The centers are subject to different zoning than the surrounding neighborhood and are considered targets for dense development.  These areas tend to have 

## Data Collection

Population data was retrieved from the city of Seattle website on December 5.  The population data is US census data aggregated into city-designated areas.  

Population of children in each given Urban Center was gleaned from city reports retrieved on December 5.  These reports are based on early results from the 2020 census, aggregated by city-designated urban centers.  


In [1]:
#imports
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt

In [8]:
# get population data
pop = pd.read_csv("./data/citydata/Urban_Centers_and_Villages_with_PL_94-171_Redistricting__Data_for_1990-2020.csv")
pop.set_index('OBJECTID', inplace = True)
pop.head()

NEIGH_NO               NEIGH_NAME UV_TYPE  \
OBJECTID                                              
1              106         Upper Queen Anne     RUV   
2              302                  Fremont     HUV   
3              205    West Seattle Junction     HUV   
4              111            Columbia City     RUV   
5              414  First Hill/Capitol Hill      UC   

                          TYPE_NAME  AREA_ACRES  F1990_PL_data_TOT_POP  \
OBJECTID                                                                 
1         Residential Urban Village   52.658720                   1921   
2                 Hub Urban Village  214.511083                   3153   
3                 Hub Urban Village  270.417026                   3627   
4         Residential Urban Village  334.543340                   4281   
5                      Urban Center  916.221801                  28940   

          F1990_PL_data_WHITE_NOT  F1990_PL_data_BLACK_NOT  \
OBJECTID                                                     
1                            1715                       56   
2                            2697                       90   
3                            3169                      145   
4                             898                     1827   
5                           21504                     3037   

          F1990_PL_data_AMI_NOT  F1990_PL_data_ASIAN_NOT  ...  \
OBJECTID                                                  ...   
1                            11                       71  ...   
2                            64                      189  ...   
3                            33                      150  ...   
4                           111                     1269  ...   
5                           534                     2403  ...   

          F1990_PL_data_GQ_I  F1990_PL_data_GQ_N_I  F2000_PL_data_GQ_I  \
OBJECTID                                                                 
1                          0                    37                   0   
2                          0                    47                   0   
3                        401                    14                 358   
4                          0                     5                  98   
5                        660                  1370                 782   

          F2000_PL_data_GQ_N_I  F2010_PL_data_GQ_I  F2010_PL_data_GQ_N_I  \
OBJECTID                                                                   
1                            0                   0                     4   
2                           63                   0                    90   
3                           13                 272                     0   
4                            7                  95                    20   
5                         2278                 332                  2329   

          F2020_PL_data_GQ_I  F2020_PL_data_GQ_N_I   Shape__Area  \
OBJECTID                                                           
1                          0                     0  2.293805e+06   
2                          0                    66  9.344065e+06   
3                        235                    12  1.177932e+07   
4                        102                    80  1.457265e+07   
5                        846                  3135  3.991046e+07   

          Shape__Length  
OBJECTID                 
1           11354.22008  
2           16236.22258  
3           20674.93070  
4           25210.67846  
5           31324.33554  

[5 rows x 110 columns]

In [9]:
pop.shape

(32, 110)

In [15]:
pop.isna().sum().sum()

0

The data has no nulls.  We have 110 columns for each of the 32 Urban Centers

## Feature Engineering

Adding a `change` feature that represents the change in population from 2010 to 2020 and a `perc_change` that represents the percentage change in population from 2010 to 2020

In [18]:
pop['change'] = pop['F2020_PL_data_TOT_POP'] - pop['F2010_PL_data_TOT_POP']

In [20]:
pop['perc_change'] = (pop['F2020_PL_data_TOT_POP'] - pop['F2010_PL_data_TOT_POP'])/pop['F2010_PL_data_TOT_POP']

In [28]:
pop[['NEIGH_NAME', 'AREA_ACRES','F2010_PL_data_TOT_POP','F2020_PL_data_TOT_POP', 'change', 'perc_change']].sort_values('perc_change', ascending=False)

NEIGH_NAME   AREA_ACRES  F2010_PL_data_TOT_POP  \
OBJECTID                                                                  
7                  South Lake Union   339.886927                   3774   
3             West Seattle Junction   270.417026                   5043   
6                         Roosevelt   170.211776                   2616   
12                 Greater Duwamish  4953.281793                   1064   
4                     Columbia City   334.543340                   4609   
20                           Uptown   333.000698                   7300   
19                         Downtown   953.281747                  26844   
30                          Ballard   495.476305                  11440   
2                           Fremont   214.511083                   3960   
13                   Madison-Miller   145.323420                   4066   
5           First Hill/Capitol Hill   916.221801                  35892   
14                       Green Lake   107.444638                   2904   
23              Bitter Lake Village   363.779566                   4519   
8               University District   751.963878                  22875   
11                        Lake City   142.267043                   3899   
22             23rd & Union-Jackson   625.266413                  10460   
25                          Admiral    98.248351                   1528   
24                         Mt Baker   491.018058                   5325   
1                  Upper Queen Anne    52.658720                   2143   
32                          Othello   498.963241                   9330   
15                        Northgate   412.291123                   6620   
28                         Eastlake   199.255956                   5084   
29                      Wallingford   258.005613                   5425   
18          Greenwood-Phinney Ridge    94.157645                   2927   
21                    Rainier Beach   346.143874                   4094   
9                        Crown Hill   270.988754                   4191   
17        Ballard-Interbay-Northend   931.972864                   1598   
27                  Morgan Junction   113.309377                   2046   
26            Aurora-Licton Springs   326.958599                   6179   
16                North Beacon Hill   266.876291                   5217   
10           Westwood-Highland Park   275.062196                   4606   
31                       South Park   263.798620                   3448   

          F2020_PL_data_TOT_POP  change  perc_change  
OBJECTID                                              
7                         12905    9131     2.419449  
3                          9160    4117     0.816379  
6                          4678    2062     0.788226  
12                         1836     772     0.725564  
4                          7661    3052     0.662183  
20                        11840    4540     0.621918  
19                        41875   15031     0.559939  
30                        17707    6267     0.547815  
2                          5841    1881     0.475000  
13                         5611    1545     0.379980  
5                         49174   13282     0.370055  
14                         3934    1030     0.354683  
23                         6100    1581     0.349856  
8                         29887    7012     0.306536  
11                         5081    1182     0.303155  
22                        13610    3150     0.301147  
25                         1978     450     0.294503  
24                         6777    1452     0.272676  
1                          2604     461     0.215119  
32                        11257    1927     0.206538  
15                         7950    1330     0.200906  
28                         6100    1016     0.199843  
29                         6506    1081     0.199263  
18                         3492     565     0.193030  
21                         4830     736     